tracecontext/src/test/java/brave/propagation/tracecontext/TraceContextPropagationTest.java

```diff
@@ -93,10 +115,65 @@
     request.put("traceparent", validTraceparent);
     request.put("tracestate", otherState + ",b3=" + validB3Single);
 
-    sampledContext = propagation.decorate(sampledContext);
-    TracestateFormat.INSTANCE.parseInto(otherState, sampledContext.findExtra(Tracestate.class));
+    TraceContext withTracestate =
+      sampledContext.toBuilder().addExtra(new Tracestate(otherState)).build();
 
     assertThat(extractor.extract(request))
-      .isEqualTo(TraceContextOrSamplingFlags.create(sampledContext));
+      .usingRecursiveComparison()
+      .isEqualTo(TraceContextOrSamplingFlags.create(withTracestate));
+  }
+
+  @Test void logOrThrow_logs() {
```

Comment:
inlined a copy of Platform.log rather than shade Platform only to defer the logger